lib/screen/feralfile_home/artwork_view.dart

```diff
@@ -334,36 +338,38 @@ class _SeriesViewState extends State<SeriesView> {
         ),
       );
 
-  void _gotoSeriesDetails(BuildContext context, FFSeries series) {
-    unawaited(
-      Navigator.of(context).pushNamed(
+  Future<void> _gotoSeriesDetails(BuildContext context, FFSeries series) async {
+    if (series.isSingle) {
+      setState(() {
+        _navigating = true;
+      });
+      try {
+        final artwork = await injector<FeralFileService>()
+            .getFirstViewableArtwork(series.id);
+        if (artwork != null) {
+          if (context.mounted) {
+            unawaited(Navigator.of(context).pushNamed(
+              AppRouter.ffArtworkPreviewPage,
+              arguments: FeralFileArtworkPreviewPagePayload(
+                artwork: artwork.copyWith(series: series),
+              ),
+            ));
+          }
+        }
+      } finally {
```

Comment:
please add cache and add sentry. Also make it to series page if any error.

Reply:
what kind of error are your thinking of?
we've never seen any error of this api?
if it's a network issue, it's pointless to navigate to series page
Beside we should not navigate to elsewhere just because it fails to go to intended page.
The purpose of this flow is to allow user see the artwork, if the artwork is not available, let them stay there